src/Doctrine/Instantiator/Instantiator.php

```diff
@@ -186,6 +187,10 @@ private function attemptInstantiationViaUnSerialization(ReflectionClass $reflect
             restore_error_handler();
 
             throw UnexpectedValueException::fromSerializationTriggeredException($reflectionClass, $exception);
+        } catch (Throwable $exception) {
+            restore_error_handler();
```

Comment:
Shame this repo doesn't support the finally statement (since php 5.3/5.4 is allowed). We could have used that to ensure `restore_error_handler` was called. ;(